panels/updates/cc-updates-panel.c

```diff
@@ -188,17 +191,23 @@ store_automatic_updates_setting (NMConnection *connection,
   if (error)
     {
       g_warning ("Error storing "NM_SETTING_TARIFF": %s", error->message);
-      return;
+      g_clear_error (&error);
+      errored = TRUE;
     }
 
-  /* Make sure the application does not exit before the callback is called */
-  g_application_hold (g_application_get_default ());
 
-  nm_remote_connection_commit_changes_async (NM_REMOTE_CONNECTION (connection),
-                                             TRUE, /* save to disk */
-                                             NULL,
-                                             on_network_changes_committed_cb,
-                                             NULL);
+  /* Only commit the changes if there were no errors. */
+  if (!errored)
+    {
```

Comment:
This is just a matter of style, but I prefer early returns.

Reply:
I did this to avoid having the `g_signal_handler_unblock()` calls in every early return (or in a `goto error` block).

Reply:
What I'm saying is that I'd prefer a
```c
if (errored)
  return;
```

here

Reply:
Oh, I see. I can do that.